app/assets/javascripts/discourse/templates/modal/insert-hyperlink.hbs

```diff
@@ -1,6 +1,18 @@
 {{#d-modal-body title="composer.link_dialog_title" class="insert-link"}}
     <div class="inputs">
       {{text-field value=linkUrl placeholderKey="composer.link_url_placeholder" class="link-url"}}
```

Comment:
this is the issue here for the observer, you are basically relying on mutation of linkUrl, instead of having an event when the textfield is changed, you could have an event on input and make `linkUrl` readonly, this ways it would make the whole code better.

I realise this is not your code at the beginning, but you are now impacted by this "bad" patter :p